### DISCUSSION TITLE
Encryption[RTSPS] mode not working

### DISCUSSION THREAD

**Discussion by madhu7021** [bluenviron/mediamtx#830] [Questions]
## Which version are you using?

v0.17.3

## Which operating system are you using?

OS

- [ x] Linux
- [ ] Windows
- [ ] macOS

Architecture

- [ x] amd64
- [ ] arm64
- [ ] arm7
- [ ] arm6

## Describe the issue

 we quickly followed the guidelines and executed the instructions. In Normal mode(RTSP) things are running fine. able to stream the data seamlessly.

But somehow Encryption mode(RTSPS) is not working. captured the details.

Followed the procedure and created RSA private key and self-signed certificate files and placed them in the path, server is running.

openssl genrsa -out server.key 2048
openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650

Enabled encryption flags in the yml file

protocols: [tcp]
encryption: optional
serverKey: server.key
serverCert: server.crt

Here the observations

**RTSP Server**

mananthula@IM-LP-528:~/Downloads/rtsp-simple-server_v0.17.3_linux_amd64$ ./rtsp-simple-server
2021/09/20 10:43:12 I [0/0] rtsp-simple-server v0.17.3
2021/09/20 10:43:12 I [0/0] [RTSP] TCP listener opened on :8554
**2021/09/20 10:43:12 I [0/0] [RTSPS] TCP listener opened on :8555**
2021/09/20 10:43:12 I [0/0] [RTMP] listener opened on :1935
2021/09/20 10:43:12 I [0/0] [HLS] listener opened on :8888
2021/09/20 10:43:32 I [0/0] [RTSPS] [conn 127.0.0.1:38760] opened
2021/09/20 10:43:32 I [0/0] [RTSPS] [conn 127.0.0.1:38760] closed
2021/09/20 10:43:45 I [0/0] [RTSPS] [conn 127.0.0.1:38762] opened
2021/09/20 10:43:45 I [0/0] [RTSPS] [conn 127.0.0.1:38762] closed

**FFMPEG**

tried with ffmpeg clients for validation

ffmpeg version is on the system is 2.8.17-0ubuntu0.1, Desktop OS is ubuntu 16.04 LTS

**Publisher**
 
mananthula@IM-LP-528:~$ ffmpeg -f v4l2 -i /dev/video0 -f rtsp rtsps://localhost:8555/mystream

the error is : Could not write header for output file #0 (incorrect codec parameters ?): Operation not permitted

**Subscriber**

mananthula@IM-LP-528:~$ ffplay -i rtsps://localhost:8555/mystream

the error is : rtsps://localhost:8555/mystream: Operation not permitted

## Describe how to replicate the issue

1. start the server
2. publish with ...
3. read with ...

## Did you attach a configuration files?

yes it is easily reproducible, direct case

**Comment by aler9** [comment]
Hello, i tested all the commands you posted and they worked flawlessly.

in my opinion, the problem can be one of the following:

1. Your camera is using an unsupported codec, in this case you can re-encode it stream before publishing:
   ```
   ffmpeg -f v4l2 -i /dev/video0 -f rtsp -c:v libx264 -tune zerolatency rtsps://localhost:8555/mystream
   ```
2. The server can't be reached due to firewall / ipv6 / other machine limitations.  Make sure that something appears in the server log when you run `ffplay` / `ffmpeg`

**Comment by saket424** [comment]
A relatively simple patch to mpv player is now available in ubuntu22.04 to support rtsps playback
https://github.com/mpv-player/mpv/issues/8480

vlc support for rtsps doesn't appear imminent anytime soon

https://forum.videolan.org/viewtopic.php?t=147135

**Comment by saket424** [comment]
It appears that all that is needed for vlc to work is for the maintainers to Compile ffmpeg with openssl support as mentioned in this link below 

https://code.videolan.org/videolan/vlc-android/-/issues/583

[a comment by an automated account is not shown]
